src/mesh/forest/cell_center_offsets.hpp

```diff
@@ -0,0 +1,139 @@
+//========================================================================================
+// (C) (or copyright) 2023. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+#ifndef MESH_FOREST_CELL_CENTER_OFFSETS_HPP_
+#define MESH_FOREST_CELL_CENTER_OFFSETS_HPP_
+
+#include <array>
+#include <map>
+#include <memory>
+#include <set>
+#include <tuple>
+#include <unordered_map>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+
+#include "basic_types.hpp"
+#include "defs.hpp"
+#include "mesh/logical_location.hpp"
+#include "utils/bit_hacks.hpp"
+#include "utils/indexer.hpp"
+
+namespace parthenon {
+namespace forest {
+enum class Direction : uint { I = 0, J = 1, K = 2 };
+
+// CellCentOffsets defines the position of a topological element
+// within a cell or a neighboring cell via offsets from the cell center. The center of
+// cell is defined by zero offsets in each direction. The faces have
+// a Offset::Low or Offset::Up in one direction and Offset::Middle
+// in the others, etc. The topological position of an element in the
+// is turned into an unsigned index 0...27 via
+// (x_offset + 1) + 3 * (y_offset + 1) + 9 * (z_offset + 1)
+
+// TODO(LFR): Consider switching this to C-style enum within a namespace to avoid
+// static_cast
+enum class Offset : int { Low = -1, Middle = 0, Up = 1 };
```

Comment:
Further up in the bvals, the ordering of offset was 0, -1, 1 IIRC.
I realize these are enums here, so it shouldn't matter, but I'm wondering if there any danger for confusion mid/long term wrt to ordering convention.

Reply:
I am not sure I see where the ordering is different?